sw/device/lib/testing/check.h

```diff
@@ -45,4 +46,28 @@
     }                                                     \
   } while (false)
 
+/**
+ * Compare `num_items` of `got` against `expected` buffer.
+ *
+ * Prints differences between `got` and `expected` before generating a test
+ * assertion if a mismatch is found.
+ *
+ * `sizeof(got[0])` is used to determine the size of each item in the buffer.
```

Comment:
Maybe this should be a comment in the macro itself?

Reply:
Done.